presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1714,13 +1715,13 @@ else if (expression instanceof DereferenceExpression) {
                     }
                 }
 
-                if (!alias.isPresent()) {
+                if (!fieldName.isPresent()) {
                     if (name != null) {
-                        alias = Optional.of(getLast(name.getOriginalParts()));
+                        fieldName = Optional.of(getLast(name.getOriginalParts()));
                     }
                 }
 
-                outputFields.add(Field.newUnqualified(alias, analysis.getType(expression), qualifiedOriginTable, alias.isPresent())); // TODO don't use analysis as a side-channel. Use outputExpressions to look up the type
+                outputFields.add(Field.newUnqualified(fieldName, analysis.getType(expression), qualifiedOriginTable, isAliased)); // TODO don't use analysis as a side-channel. Use outputExpressions to look up the type
```

Comment:
you had a good point about the method being confusing.  I'm going to change this to call column.getAlias().isPresent() right here so it's clear where it's coming from.